idaes/tests/performance/test_models.py

```diff
@@ -0,0 +1,74 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Author: Andrew Lee
+"""
+import pytest
+import gc
+
+import pyomo.common.unittest as unittest
+from pyomo.util.check_units import assert_units_consistent
+from pyomo.common.timing import TicTocTimer
+
+from idaes.models.properties.modular_properties.examples.tests import test_HC_PR
+from idaes.models.unit_models.tests import test_heat_exchanger_1D
+
+
+class TestModel(unittest.TestCase):
+    def recordData(self, name, value):
+        """A method for recording data associated with a test.
+
+        This method is only meaningful when running this TestCase with
+        'nose', using the TestData plugin.
+
+        """
+        tmp = getattr(self, "testdata", None)
+        if not tmp is None:
+            tmp[name] = value
+
+    def _run_test(self, model_lib, init_model, solve_model):
+        gc.collect()
+        timer = TicTocTimer()
+
+        model = model_lib()
+        self.recordData("build model", timer.toc("build model"))
+
+        gc.collect()
+        timer.tic(None)
+        assert_units_consistent(model)
```

Comment:
This call to `assert_units_consistent` seems to ensure that tests that build a dynamic model will fail unless the time discretization transformation is performed as part of `init_model`. Is this intentional? I would prefer time discretization to be either part of "build model" or its own category, but don't feel that strongly (for now) about it being lumped in with "initialize".

Reply:
We had not thought about this issue, and ultimately this is not an IDAES issue but rather a Pyomo issue that needs to be fixed (and we need to keep prodding them to get it elevated in priority). However, also see more more general response below.